File: backend/app/schemas/resume.py
import uuid
from datetime import datetime

from pydantic import BaseModel


# Request body for creating a resume: just the label and file's content type, sent before the actual file bytes exist anywhere.
class ResumeCreate(BaseModel):
    label: str
    content_type: str


# Request body for updating a resume's label and/or marking it as the base resume.
class ResumeUpdate(BaseModel):
    label: str | None = None
    is_base: bool | None = None


# Response shape for a resume, built from the ORM object via from_attributes.
class ResumeRead(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    label: str
    content_type: str
    is_base: bool
    created_at: datetime

    model_config = {"from_attributes": True}


# Response returned right after creating a resume: the saved record plus the presigned S3 URL the frontend uploads the actual file to.
class ResumeUploadResponse(BaseModel):
    resume: ResumeRead
    upload_url: str


# Response containing a short-lived presigned URL for viewing/downloading a resume's file.
class ResumeDownloadResponse(BaseModel):
    download_url: str

File: frontend/src/types/resume.ts
// A resume/cover letter record, mirroring the backend's ResumeRead schema.
export interface Resume {
  id: string
  user_id: string
  label: string
  content_type: string
  is_base: boolean
  created_at: string
}

// Response from POST /resumes: the saved record plus the presigned S3 URL to upload the actual file to.
export interface ResumeUploadResponse {
  resume: Resume
  upload_url: string
}

// Response from GET /resumes/{id}/download: a short-lived presigned URL for viewing the file.
export interface ResumeDownloadResponse {
  download_url: string
}

// File types the upload form accepts, matching the backend's ALLOWED_CONTENT_TYPES allowlist.
export const ALLOWED_RESUME_CONTENT_TYPES = [
  'application/pdf',
  'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
] as const

// Client-side size cap enforced before ever calling the API — not mirrored server-side, since a
// resume is never legitimately larger than this for a personal-use app.
export const MAX_RESUME_SIZE_BYTES = 10 * 1024 * 1024
